### ISSUE TITLE
EXTRACT_ALL=YES parses non doxygen comments (Origin: bugzilla #550158)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#3065]
status NEW severity *normal* in component *general* for *---*
Reported in version *1.5.6* on platform *Other*
Assigned to: Dimitri van Heesch

Original attachment names and IDs:
- _[Doxyfile](https://sourceforge.net/projects/doxygen/files/attachments/bug550158_117747_Doxyfile/download/download)_ (ID 117747)
- _[test.hpp](https://sourceforge.net/projects/doxygen/files/attachments/bug550158_117748_test.hpp/download/download)_ (ID 117748)

On 2008-09-01 08:48:07 +0000, Peter Klotz wrote:

    Please describe the problem:
    When EXTRACT_ALL is set to YES, doxygen seems to parse non doxygen comments
    and issues warnings that should not be present.
    
    The attached files test.hpp and Doxyfile show this behavior.
    
    Steps to reproduce:
    1. Just run doxygen in a directory with Doxyfile and test.hpp
    2. 
    3. 
    
    
    Actual results:
    This warning is displayed:
    
    /tmp/doxygen/test.hpp:1: Warning: argument 'p_name' of command @param is not
    found in the argument list of setFocus()
    
    Expected results:
    No warning should be displayed since activateTool() and its whole
    documentation is commented out and should therefore be invisible for
    doxygen.
    
    The documentation states that comments may start with "///" but in this case
    the comment starts with "// /**" which should be ignored.
    
    When changing the code to "//x /**" doxygen no longer displays the warning.
    
    Does this happen every time?
    Yes
    
    Other information:

On 2008-09-01 08:49:12 +0000, Peter Klotz wrote:

    Created attachment 117747
    The Doxyfile

On 2008-09-01 08:49:47 +0000, Peter Klotz wrote:

    Created attachment 117748
    A source file demonstrating the behavior

**Comment by albert-github** [comment]
In the `scanner.l` we have the rules:
```
 <FindMembers,FindFields,MemberSpec,FuncQual,SkipCurly,Operator,ClassVar,SkipInits,Bases,OldStyleArgs>("//"{B}*)?"/**"/[^/*] {
```
and
```
<DocBlock>^{B}*("//")?{B}*"*"+/[^//a-z_A-Z0-9*] { // start of a comment line
```
that cause this behavior. We see here that a doxygen comment of type `/**` can be preceded by something like `"//"{B}*`. 

Commenting out can be done by:
- removing one of the `*` from `/**`
- changing the first `//` into `// /`
- placing a `#if 0... #endif` block around the code part

The reason for this behavior / this rule is unclear (to me).

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7300.

This is an extension to the pull request #6649

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.17.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
